install/gcp/upgrade_tools/db_migrator.py

```diff
@@ -0,0 +1,113 @@
+import sys
+
+# Importing migrate.changeset adds some new methods to existing SQLAlchemy
+# objects but we will not be calling the library directly.
+import migrate.changeset  # noqa: F401, pylint: disable=unused-import
+from sqlalchemy.exc import OperationalError
+
+import google.cloud.forseti.services.scanner.dao as scanner_dao
+import google.cloud.forseti.services.inventory.storage as inventory_dao
+import google.cloud.forseti.services.dao as general_dao
+
+from google.cloud.forseti.common.util import logger
+
+
+DEFAULT_DB_CONN_STR = 'mysql://root@127.0.0.1:3306/forseti_security'
+LOGGER = logger.get_logger(__name__)
+
+
+class ColumnAction:
+    """Column action class."""
+    DROP = 'DROP'
+    CREATE = 'CREATE'
+
+
+def create_col(table, col):
+    """Create Column.
+
+    Args:
+        table (Table): The table object.
+        col (Column): The column object."""
+    LOGGER.info('Attempting to create column: %s', col.name)
+    col.create(table, populate_default=True)
+
+
+def drop_col(table, col):
+    """Create Column.
+
+    Args:
+        table (Table): The table object.
+        col (Column): The column object."""
+    LOGGER.info('Attempting to drop column: %s', col.name)
+    col.drop(table)
+
+
+ColumnActionMap = {ColumnAction.DROP: drop_col,
+                   ColumnAction.CREATE: create_col}
+
+
+def migrate_schema(engine, base):
+    """Create all tables in the database if not existing.
+
+    Args:
+        engine (object): Database engine to operate on.
+        base (Base): Declarative base.
+    """
+    # Create tables if not exists.
+    base.metadata.create_all(engine)
+    base.metadata.bind = engine
+
+    # Update schema changes.
+    # Find all the child classes inherited from declarative base class.
+    base_subclasses = _find_subclasses(base)
+
+    # Find all the Table objects for each of the classes.
+    # The format of tables is: {table_name: Table object}.
+    tables = base.metadata.tables
+
+    schema_update_method_name = 'update_schema'
+
+    for subclass in base_subclasses:
+        schema_update = getattr(subclass, schema_update_method_name, None)
+        if callable(schema_update) and subclass.__tablename__ in tables:
+            LOGGER.info('Updating table %s', subclass.__tablename__)
+            # schema_update will require the Table object.
+            try:
+                table = tables.get(subclass.__tablename__)
+                columnMapping = schema_update()
+                for columnAction, column in columnMapping.iteritems():
+                    columnAction = columnAction.upper()
+                    if columnAction in ColumnActionMap:
+                        ColumnActionMap.get(columnAction)(table, column)
```

Comment:
Should there be a `try/except` around this line, so that if a column exists, then it can move to the next column?

Reply:
Updated.